File: src/API/src/Modules/Products/Products.Application/ViewModels/ProductListItemViewModel.cs
namespace Products.Application.ViewModels
{
    public class ProductListItemViewModel
    {
        public Guid Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public Guid ProductTypeId { get; set; }
        public string ProductTypeName { get; set; } = string.Empty;
        public Guid DeploymentTypeId { get; set; }
        public string DeploymentTypeName { get; set; } = string.Empty;
        public Guid OwnershipTypeId { get; set; }
        public string OwnershipTypeName { get; set; } = string.Empty;
        public Guid? OwnerPartnerId { get; set; }
        public string? OwnerPartnerName { get; set; }
        public bool IsSubscriptionBased { get; set; }
        public bool IsLicenseBased { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: src/ClientApp/src/app/pages/products/view-models/product-list-item.view-model.ts
export interface ProductListItemViewModel {
    id: string;
    code: string;
    name: string;
    productTypeId: string;
    productTypeName: string;
    deploymentTypeId: string;
    deploymentTypeName: string;
    ownershipTypeId: string;
    ownershipTypeName: string;
    ownerPartnerId?: string | null;
    ownerPartnerName?: string | null;
    description?: string | null;
    isSubscriptionBased: boolean;
    isLicenseBased: boolean;
    isActive: boolean;
    createdAt: string;
}
